src/libs/ReportUtils.ts

```diff
@@ -3459,14 +3473,78 @@ function shouldHideReport(report: OnyxEntry<Report>, currentReportId: string): b
     return parentReport?.reportID !== report?.reportID && !isChildReportHasComment;
 }
 
+/**
+ *  Checks if there are any violations belonging to the transaction in the transactionsViolations Onyx object
+ *  then checks that the violation is of the proper type
+ */
+function transactionHasViolation(transactionID: string, transactionViolations?: TransactionViolations): boolean {
+    const violations: TransactionViolation[] = transactionViolations ? transactionViolations[transactionID] || [] : [];
+    if (!violations) {
+        return false;
+    }
+    return violations.some((violation: TransactionViolation) => violation.type === 'violation');
+}
+
+/**
+ *  Checks to see if a report's parentAction is a money request that contains a violation
+ *  This only applies to report submitter and for reports in the open and processing states
+ */
+
```

Comment:
We're not listing all the conditions here (and we shouldn't), so we shouldn't list submitter and report state either imo
```suggestion
 */
```